python/louis/__init__.py.in

```diff
@@ -79,32 +71,49 @@ liblouis.lou_version.restype = c_char_p
 liblouis.lou_charSize.restype = c_int
 
 liblouis.lou_translateString.argtypes = (
-    c_char_p, c_wchar_p, POINTER(c_int), c_wchar_p, 
+    c_char_p, POINTER(c_char), POINTER(c_int), POINTER(c_char), 
          POINTER(c_int), POINTER(c_ushort), POINTER(c_char), c_int)
 
 liblouis.lou_translate.argtypes = (
-    c_char_p, c_wchar_p, POINTER(c_int), c_wchar_p, 
+    c_char_p, POINTER(c_char), POINTER(c_int), POINTER(c_char), 
          POINTER(c_int), POINTER(c_ushort), POINTER(c_char), 
          POINTER(c_int), POINTER(c_int), POINTER(c_int), c_int)
 
 liblouis.lou_backTranslateString.argtypes = (
-         c_char_p, c_wchar_p, POINTER(c_int), c_wchar_p,
+         c_char_p, POINTER(c_char), POINTER(c_int), POINTER(c_char),
          POINTER(c_int), POINTER(c_ushort), POINTER(c_char), c_int)
 
 liblouis.lou_backTranslate.argtypes = (
-         c_char_p, c_wchar_p, POINTER(c_int), c_wchar_p, POINTER(c_int),
+         c_char_p, POINTER(c_char), POINTER(c_int), POINTER(c_char), POINTER(c_int),
          POINTER(c_ushort), POINTER(c_char), POINTER(c_int), POINTER(c_int),
          POINTER(c_int), c_int)
 
 liblouis.lou_hyphenate.argtypes = (
-         c_char_p, c_wchar_p, c_int, POINTER(c_char), c_int)
+         c_char_p, POINTER(c_char), c_int, POINTER(c_char), c_int)
 
 liblouis.lou_checkTable.argtypes = (c_char_p,)
 
 liblouis.lou_compileString.argtypes = (c_char_p, c_char_p)
 
 liblouis.lou_getTypeformForEmphClass.argtypes = (c_char_p, c_char_p)
 
+#{ Module Configuration
+#: Specifies the charSize (in bytes) used by liblouis.
+#: This is fetched once using L{liblouis.lou_charSize}.
+#: Call it directly, since L{charSize} is not yet defined.
+#: @type: int
+wideCharBytes = liblouis.lou_charSize()
+#: Specifies the number by which the input length should be multiplied
+#: to calculate the maximum output length.
+#: @type: int
+# This default will handle the case where every input character is
+# undefined in the translation table.
+outlenMultiplier = 4 + wideCharBytes * 2
+#: Specifies the encoding to use when converting from byte strings to unicode strings.
+#: @type: str
+conversionEncoding = "utf_%d_le" % (wideCharBytes * 8)
```

Comment:
Does this work on Python 3?

Reply:
Actually, no, but the issue I found was already there.

unicode(bytes(b"hello")) returns "b'hello'", including the literal. That's probably the way how python 3 makes a distinction between byte and unicode strings.

Reply:
[**@egli**](https://github.com/egli) commented on [31 aug. 2018 17:22 CEST](https://github.com/liblouis/liblouis/pull/633#discussion_r214389252 "2018-08-31T15:22:41Z - Replied by Github Reply Comments"):
> But if it doesn't work on Python3 how can this PR fix [#55](https://github.com/liblouis/liblouis/issues/55)?

I'm sorry, I didn't express myself clear enough.

It does work on python 3, but I expected it to work for both byte and unicode strings. It only works for unicode strings (i.e. the str type in python 3) which is the default. People who want to throw byte strings at this function should fdecode them first. Im'not even sure whether this should be considered a bug.

Reply:
Hah, I actually meant to ask if the `"string %d" % (42)` syntax was still allowed/preferred. Looks like it is.